### ISSUE TITLE
App reloading inside iFrame when renewing token - Solution 2

### ISSUE THREAD

**Issue by Tsunade90** [AzureAD/azure-activedirectory-library-for-js#366]
I'm working on the project for which that issue was posted:

https://github.com/AzureAD/azure-activedirectory-library-for-js/issues/298#issuecomment-225035980

We upgrated our adal.js version to 1.0.11 and use the workaround described here:
https://gist.github.com/tushargupta51/5fa6d000357120adbc8fd1d5c68853c4

It solves our issues in localhost but not when the code is minified/uglyfied and deployed on a distant server, the page stays stuck on frameRedirect.html when the token is trying to be renewed:
http://mysite.azurewebsites.net/frameRedirect.html#/id_token=eyJ0eXAiOiJKV1Q[...]

When debbuging the _locationChangeHandler_ (adal-angular.js file) function, it appears that the variable _loginStartPage_ is always empty (_var loginStartPage = _adal._getItem(_adal.CONSTANTS.STORAGE.LOGIN_REQUEST);_) so the user can't be redirected to the app's home page after the token has been renewed. 
If I set manually a value inside _loginStartPage_ to see what happens, the app reloads forever.

We downgraded the adal version to 1.0.7 for the moment because our app is in production environment so the user can't face a white empty page but we can't keep this version as it causes weird behaviours (the js files are loaded 5-6 times).

Thank you

**Comment by tushargupta51** [comment]
@Tsunade90 Is this issue specific to some browser? Or does it occur on all browsers: Chrome, Firefox, IE/Edge?

**Comment by Tsunade90** [comment]
It occurs on all browsers, sorry for not mentioning it

**Comment by Tsunade90** [comment]
If it helps: seems to have something to do with the "angular components" architecture because if I don't use a redirectUri in my adal config, I get the error:

Error: [$compile:tpload] Failed to load template: scripts/components/master/master-page.html (HTTP status: undefined undefined)
http://errors.angularjs.org/1.5.7/$compile/tpload?p0=scripts%2Fcomponents%2Fmaster%2Fmaster-page.html&p1=undefined&p2=undefined

index.html:
<body>
   <my-main-component></my-main-component>

**Comment by tushargupta51** [comment]
@Tsunade90 Sorry for the delay in getting back on this. Can you please share the special frame html that you have, adal init configuration and route configuration? That will help me debug this further. Also, console logs will help too. You can turn on logging by adding this globally:

Logging = {
    level: 3,
    log: function (message) {
        console.log(message);
    }
};

> Error: [$compile:tpload] Failed to load template: scripts/components/master/master-page.html (HTTP status: undefined undefined)

This error can occur if user is not logged in and master-page.html is neither specified as templateUrl of one of the anonymous routes or added in anonymousEndpoints array.

**Comment by tberst** [comment]
You can find the log below : 
[verbose_logs.txt](https://github.com/AzureAD/azure-activedirectory-library-for-js/files/482706/verbose_logs.txt)

I've followed the path that leads to an empty adal.login.request. Apparently, when we access the app with an expired token we have a call to the authenticationContext.login() which starts a login process with the localstorage  CONSTANTS.STORAGE.LOGIN_REQUEST set to the current page (which is good).

During the same time, there is call to _renewIdToken (from I don't know where) which erase the CONSTANTS.STORAGE.LOGIN_REQUEST in the localstorage. 

By the time the first login process completes its thing, we are redirected to the frameredirect.html with an empty CONSTANTS.STORAGE.LOGIN_REQUEST in the localstorage.

**Comment by tushargupta51** [comment]
@tberst Thanks for debugging into this. The library should not be resettting the login_request value when renewing tokens.
@rohitnarula7176 Can you please fix this in next version? We don't need save statements in these 2 lines:
https://github.com/AzureAD/azure-activedirectory-library-for-js/blob/1.0.11/lib/adal.js#L309
https://github.com/AzureAD/azure-activedirectory-library-for-js/blob/1.0.11/lib/adal.js#L284

**Comment by tushargupta51** [comment]
This is fixed in 1.0.13. Let us know if you still running into this issue.

**Comment by malay2012** [comment]
We have upgraded to version 1.0.13, but still have this issue for some Edge browser. we have the issue in earlier version as well. This is happening in initial login itself. We have cleared the browser cache and tried fresh.

a. We have observed that initially in sessionstorage values are stored for clientId, but there seems no calls for api resource. Then continuosly failing even though we forced login in place.
b. After clearing cache tried to make fresh call then onwards it is continously calling authorize with Id_token in the URL.

We will try to get some logs, this is not happening in local dev environment.
